### ISSUE TITLE
Unable to print some variables in gdb (type information missing)

### ISSUE THREAD

**Issue by gusain** [llvm/llvm-project#60088]
I'm using Clang 15.0.7 to compile (with gcc tool chain 11.2) and llvm lld to link. When I load binary, for some of the symbols, type information is missing and I see <incomplete type> in gdb.

Interesting enough, if I do 'list <file where type is defined>` in gdb and then retry, it works as expected and no more incomplete type message. 
Only reason for filling it against llvm is that when I compile same source with gcc and gold linker, I don't run into this problem. I'm not sure where problem is actually.

**Comment by asl** [comment]
@gusain Will you be able to share a reproducer?

**Comment by gusain** [comment]
No, I couldn't reproduce it on small app (. I can share binary debuginfo if it helps (tell me exact commands I need to run to extract it)

**Comment by dwblaikie** [comment]
We'd really need a reproducer to investigate this well - knowing what DWARF was produced is probably still not enough to necessarily figure out what the source code looks like/why the compiler produced that DWARF.

**Comment by gusain** [comment]
I'm not sure if/how I can help because it's a large code base and when I try to create reduced version, it seems to work fine.

**Comment by dwblaikie** [comment]
Less chance we can figure it out without any repro steps than you - but I'll try to give some pointers to help.

It's mostly, basically "reduce your program incrementally until it's so simple there's nothing left" - eg: try making the "main" function in your program do the thing you wan to debug/doesn't work correctly (declare a variable of some interesting type there, see what happens when you step into main) - if that doesn't reproduce the problem, then simplify main to call directly (as directly as you can) into the function that does have a variable that seems to have the problem, remove as much other stuff as you can once you've got that line from main to the debugging point as short/simple as possible - remove properties of the type (member functions/variables, base classes, etc) as much as you can while still producing the problematic debugger behavior.

(check if it works with gcc built code? if it's specifically a difference between gcc and clang, maybe add that to your test script so you can easily check that you've maintained that invariant as you simplify the code)

**Comment by gusain** [comment]
Yes, it works with gcc as I mentioned before.
